### ISSUE TITLE
Autoscaling Up Wait Time Needed

### ISSUE THREAD

**Issue by ahaverbuch** [LeanerCloud/AutoSpotting#103]
# Github issue #

## Issue type ##

Feature Idea

## Summary ##

We are seeing an issue on autoscaling groups that are  autoscaling while autospotting is running.

Two fixes would help - this covers one of them: (#104 is the other)
1) Have a MIN_TIME_AFTER_SCALING variable to wait out any autoscaling event and let things scale and stabilize before trying to replace into spot, especially on scale up. The lack of this is getting new machines dropped too quickly during scale up events.

**Comment by cristim** [comment]
What's the impact on the autoscaling group? 

I would rather use the grace period of the group instead of adding such a configuration option.

The algorithm would skip the run if the instance to be replaced is still in the grace period.

What do you think?

**Comment by ahaverbuch** [comment]
Grace period might work. But my concern is heavy traffic periods on scale up. We sometimes see periods where we might need to scale a lot in one policy, then a few minutes later, scale again. The grace period for a single machine has passed, but by swapping out nodes during this due to autospotting, we end up changing various metrics we use for scaling, and impacting the overall throughput of the cluster.

Given the autospotting goal of working with long-term machines leading to spot, scaling events and anywhere around it seems like exactly the period of time where the code should remain dormant. It's similar to how scaling policies can have a wait on them to block other scaling events from occurring; that's a different number you might want from a given machines grace period.

**Comment by cristim** [comment]
I see, that totally makes sense if you mention multiple scaling actions happening in short time.

We could use the same grace period interval, but since the last scaling activity happening in the group, regardless of the uptime of individual on-demand instances.

**Comment by ahaverbuch** [comment]
@cristim hope all is well; note: this issue is 5+ years old, feel free to close (I am no longer involved in the project at issue)

**Comment by cristim** [comment]
Thanks @ahaverbuch, actually this should actually no longer be an issue for the last year or so.

The current version replaces instances immediately after launch without them having time to boot up.

But this is actually a good time to look at previous issues and close whatever is no longer relevant and see what still makes sense to build next.
